### ISSUE TITLE
Obtain id of enum variant at "runtime"

### ISSUE THREAD

**Issue by passcod** [sharksforarms/deku#141]
It would be nice to be able to obtain the deku `id` of a given enum variant from a method on the enum, for testing and other purposes (custom writers?). E.g.

```rust
#[derive(DekuRead)]
#[deku(type = "u8")]
enum Request {
	#[id = "0x01"]
	Cats { toy: Toy },

	#[id = "0x10"]
	Dogs { ball: Ball },
}

fn test_request() {
	let request = vec![Request::Cats { toy }.deku_id()];
	Request::from_bytes((&request, 0))?; // ...
}
```

Currently I make do with duplicating the IDs in a match in a method.

**Comment by wcampbell0x2a** [comment]
You may want to look at: https://github.com/sharksforarms/deku/pull/139

**Comment by passcod** [comment]
That won't work with non-unit enums, will it?

**Comment by wcampbell0x2a** [comment]
> That won't work with non-unit enums, will it?

True

**Comment by wcampbell0x2a** [comment]
@passcod Current there is a MR(https://github.com/sharksforarms/deku/pull/176) for this, if you have any input lmk

**Comment by sharksforarms** [comment]
Now available in 0.11.0
